cytominer_eval/operations/util.py

```diff
@@ -131,3 +133,87 @@ def get_grit_entry(df: pd.DataFrame, col: str) -> str:
         len(entries.unique()) == 1
     ), "grit is calculated for each perturbation independently"
     return str(list(entries)[0])
+
+
+class DistributionEstimator:
+    def __init__(self, arr):
+        self.sigma = EmpiricalCovariance().fit(arr)
+        
+    def mahalanobis(self, X):
+        return(self.sigma.mahalanobis(X))
+
+
+def calculate_mahalanobis(
+    pert_df: pd.DataFrame, control_df: pd.DataFrame
+    ) -> pd.Series:
+    """
+    Usage: Designed to be called within a pandas.DataFrame().groupby().apply()
+    """
+    assert len(control_df) > 1, "Error! No control perturbations found."
+        
+    # Get dispersion and center estimators for the 
+    control_estimators = DistributionEstimator(control_df)
+    
+    # Distance between mean of perturbation and control
+    maha = control_estimators.mahalanobis(
+                np.array(np.mean(pert_df, 0)).reshape(1, -1))[0]
```

Comment:
`DistributionEstimator().mahalanobis()` (which is basically a wrapper of `sklearn.covariance.EmpiricalCovariance.mahalanobis`) can compute distances for multiple points within  a single call. Here we subset it because we compute the distance for a single point and need to access this single value. I think it makes more sense to do it here rather than removing a feature of `DistributionEstimator().mahalanobis()`.